kotlin/internal/jvm/jvm.bzl

```diff
@@ -198,9 +198,17 @@ _common_attr = utils.add_dicts(
         [Attributes common to all build rules](https://docs.bazel.build/versions/master/be/common-definitions.html#common-attributes).""",
             allow_files = True,
         ),
+        "associates": attr.label_list(
```

Comment:
any idea how this will interact with the intellij stuff? the aspect has a list of attributes it walks and this is introducing something new

[https://github.com/bazelbuild/intellij/blob/master/aspect/intellij_info_impl.bzl](url)

Reply:
As I understand it, the deps are extracted from the actual providers, which in this context include the associates as deps - so they should just show up. That said, I've done some initial testing (I used the intellij plugin myself) and haven't seen any issues with it figuring things out, but I've not used it (yet) in complicated scenarios, so we'll have to test it more.

In a pinch, we can make the IJ stuff more aware of associates, but it oughtn't need that, since it's using the underlying providers, and those providers include deps+associates in their JavaInfo dependency information.

Reply:
)*$%)(*^&)!

`associates` will not be included in a very precise edge case -- e.g. if the only included target references a set of dependencies by `associates`, it will not be seen due to the aspect only walking certain attributes.

Walked attributes: 
Definitions: https://github.com/bazelbuild/intellij/blob/master/aspect/intellij_info_impl.bzl#L29
Usage: https://github.com/bazelbuild/intellij/blob/master/aspect/intellij_info_impl.bzl#L1090

Reply:
It's the only argument define associates in two places I've seen.

Reply:
Approving  -- it's a pretty small edge case (if really annoying). Most usage is to include entire subdirs. I believe we'd need to make a request against the plugin.

Reply:
Ugh. I had a whole theory of why it would work, but in reading the code, I see it doesn't do what I thought it did. Yeah, damn.

Reply:
I'm going to merge this, and we can sort it out, since the problem will only occur if one uses the feature (and the status quo is that the feature isn't used), and as you say, it's a small edge-case.